src/wp-includes/rest-api.php

```diff
@@ -1814,6 +1876,54 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 		}
 	}
 
+	if ( isset( $args['anyOf'] ) ) {
+		$is_valid_found = false;
+		$errors         = array();
+
+		foreach ( $args['anyOf'] as $schema ) {
+			$is_valid = rest_validate_value_from_schema( $value, $schema, $param );
+			if ( ! is_wp_error( $is_valid ) ) {
+				$is_valid_found = true;
+				break;
+			}
+
+			$errors[] = array(
+				'error_object' => $is_valid,
+				'schema'       => $schema,
+			);
+		}
+
+		if ( ! $is_valid_found ) {
+			return rest_get_combining_operation_error( $value, $param, $errors );
+		}
+	}
+
+	if ( isset( $args['oneOf'] ) ) {
+		$is_valid_found = false;
+		$errors         = array();
+
+		foreach ( $args['oneOf'] as $schema ) {
+			$is_valid = rest_validate_value_from_schema( $value, $schema, $param );
+			if ( ! is_wp_error( $is_valid ) ) {
+				if ( $is_valid_found ) {
+					/* translators: 1: Parameter. */
+					return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s matches more than one schema.' ), $param ) );
```

Comment:
Let's try and include the schemas that it matches against in the error message when possible. We can look for a `title` for each schema and then combine them using `wp_sprintf` and the `%l` directive.

And then in the error data we should always include the position of the failed schemas.